nodes/vp_track_node.cpp

```diff
@@ -27,6 +30,163 @@ namespace vp_nodes {
+				insert_iterator<set<int>>(unmatchedDetections, unmatchedDetections.begin()));
+		}
+		else if (detNum < trkNum) // there are unmatched trajectory/predictions
+		{
+			for (unsigned int i = 0; i < trkNum; ++i)
+				if (assignment[i] == -1) // unassigned label will be set as -1 in the assignment algorithm
+					unmatchedTrajectories.insert(i);
+		}
+		else
+		{}
+        // filter out matched with low IOU
+		matchedPairs.clear();
+		for (unsigned int i = 0; i < trkNum; ++i)
+		{
+			if (assignment[i] == -1) // pass over invalid values
+				continue;
+			if (1 - iouMatrix[i][assignment[i]] < iouThreshold)
+			{
+				unmatchedTrajectories.insert(i);
+				unmatchedDetections.insert(assignment[i]);
+			}
+			else
+				matchedPairs.push_back(cv::Point(i, assignment[i]));
+		}
+        // 3.3. updating trackers
+		// update matched trackers with assigned detections.
+		// each prediction is corresponding to a tracker
+		int detIdx, trkIdx;
+		for (unsigned int i = 0; i < matchedPairs.size(); i++)
+		{
+			trkIdx = matchedPairs[i].x;
+			detIdx = matchedPairs[i].y;
+			trackers[trkIdx].update(cv::Rect_<float>(meta->face_targets[detIdx]->x, meta->face_targets[detIdx]->y, meta->face_targets[detIdx]->width, meta->face_targets[detIdx]->height));
+		}
+
+		// create and initialise new trackers for unmatched detections
+		for (auto umd : unmatchedDetections)
+		{
+			KalmanTracker tracker = KalmanTracker(cv::Rect_<float>(meta->face_targets[umd]->x, meta->face_targets[umd]->y, meta->face_targets[umd]->width, meta->face_targets[umd]->height));
+			trackers.push_back(tracker);
+		}
+
+        // get trackers' output
+		frameTrackingResult.clear();
+		for (auto it = trackers.begin(); it != trackers.end();)
+		{
+			if (((*it).m_time_since_update < 1) &&
+				((*it).m_hit_streak >= min_hits))
+			{
+				TrackingBox res;
+				res.box = (*it).get_state();
+				res.id = (*it).m_id + 1;
+				res.frame = meta->frame_index;
+				frameTrackingResult.push_back(res);
+				it++;
+			}
+			else
+				it++;
+
+			// remove dead tracklet
+			if (it != trackers.end() && (*it).m_time_since_update > max_age)
+				it = trackers.erase(it);
+		}
+
+
+        // ��tracker ���� vp_frame_targets
+        for (const auto tb : frameTrackingResult)
+        {
+            auto vp_element = std::make_shared<vp_objects::vp_frame_element>(tb.id);
+            auto vp_target = std::make_shared<vp_objects::vp_frame_target>(tb.box.x, tb.box.y, tb.box.width, tb.box.height, 0, 1, tb.frame, meta->channel_index);
+            //std::vector<std::tuple<std::shared_ptr<vp_frame_element>, std::shared_ptr<vp_frame_target>, vp_ba::vp_ba_flag>> ba_flags_map;
+            vp_target->track_id = tb.id;
+            vp_target->tracks.push_back(vp_objects::vp_rect(tb.box.x, tb.box.y, tb.box.width, tb.box.height));
+			//printf("------------------------------------------id = %d\n", tb.id);
+			meta->ba_flags_map.push_back(std::make_tuple(vp_element, vp_target, vp_ba::vp_ba_flag::NONE));
```

Comment:
here is the logic of writing tracking results back to vp_frame_meta.

updating `track_id` and `tracks` field for `vp_frame_target` or `vp_frame_face_target` is enough.

you can figure out that custom logic in all nodes SHOULD follow the same rules like below:
- do something (calculating or tracking)
- write results back to vp_frame_meta.
- done